library/user_term.py

```diff
@@ -0,0 +1,56 @@
+import dataclasses
+from textwrap import dedent
+
+
+@dataclasses.dataclass()
+class UserTermState:
+    """Reset machine for next turn"""

```

Comment:
What is a "reset machine"? Maybe explain a bit more of what this is used for in the docstring?